packages/frontend/src/components/Explorer/ExportGsheets/index.tsx

```diff
@@ -25,10 +25,28 @@ const ExportGsheets: FC<ExportGsheetProps> = memo(
             health.data?.auth.google.googleDriveApiKey !== undefined;
 
         const { showToastError, showToast } = useToaster();
+        const scheduleGoogleSheetsExport = async () => {
+            try {
+                const scheduledGoogleSheetsData = await getGsheetLink();
+                const url = await pollCsvFileUrl(scheduledGoogleSheetsData);
+                if (url) window.open(url, '_blank'); //Open in new tab
 
-        const { mutateAsync: exportCsvMutation } = useMutation(
-            [],
-            () => getGsheetLink(),
+                return;
+            } catch (error) {
+                AppToaster.dismiss('exporting-gsheets');
+
+                showToastError({
+                    title: `Unable to upload Google Sheets`,
+                    subtitle: error?.error?.message,
+                });
+                throw error;
+            }
```

Comment:
waits for all of these to run so that we don't get a false `onSuccess`.

If there's an error, we have to throw it so that `react-query` can call the `onError` callback